### PR TITLE
Implement `Send-Feedback` cmdlet.

### PR DESCRIPTION
**Pull request by twitchax**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by twitchax** [comment]
I used some C# 7 features apparently...fixing.

File: src/ResourceManager/Profile/Commands.Profile/Tenant/GetAzureRMTenant.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public override void ExecuteCmdlet()
         {
             var profileClient = new RMProfileClient(AzureRmProfileProvider.Instance.Profile);
 
-            WriteObject(profileClient.ListTenants(TenantId).Select((t) => (PSAzureTenant)t), enumerateCollection: true);
+            WriteObject(profileClient.ListTenants(TenantId).Cast<PSAzureTenant>(), enumerateCollection: true);
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/Tenant/GetAzureRMTenant.cs]: This is just going to cause merge conflicts with my PR, please remove.

File: src/ResourceManager/Profile/Commands.Profile/Feedback/SendFeedback.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+            if (!this.CheckIfInteractive())
+                throw new PSInvalidOperationException(String.Format(Resources.SendFeedbackNonInteractiveMessage, nameof(SendFeedbackCommand)));
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            this.WriteQuestion(Resources.SendFeedbackRecommendationQuestion);
+            int recommendation;
+            if (!int.TryParse(this.Host.UI.ReadLine(), out recommendation) || recommendation < 0 || recommendation > 10)
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/Feedback/SendFeedback.cs]: For this, you might want to use PromptForChoice()

File: src/ResourceManager/Profile/Commands.Profile/Tenant/GetAzureRMTenant.cs (hunk 1 above)
Comment (twitchax) [inline src/ResourceManager/Profile/Commands.Profile/Tenant/GetAzureRMTenant.cs]: Done.

File: src/ResourceManager/Profile/Commands.Profile/Feedback/SendFeedback.cs (hunk 2 above)
Comment (twitchax) [inline src/ResourceManager/Profile/Commands.Profile/Feedback/SendFeedback.cs]: I tried this, and it looks really strange...not sure we want to go this route.

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

**Comment by twitchax** [comment]
@azuresdkci test this please. :)

File: src/ResourceManager/Profile/Commands.Profile/Feedback/SendFeedback.cs (hunk 2 above)
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/Feedback/SendFeedback.cs]: WOrkt a look, it might be too many choices for that to work well

**Comment by twitchax** [comment]
@azuresdkci test this please.

**Comment by twitchax** [comment]
@azuresdkci test this please.

**Comment by twitchax** [comment]
Looks like build is failing due to a lack of "playPS" on the build machine for help generation?

```
D:\workspace\powershell\build.proj(194,5): error MSB3073: The command "C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NonInteractive -NoLogo -NoProfile -Command ". D:\workspace\powershell\tools\GenerateHelp.ps1 -ValidateMarkdownHelp -GenerateMamlHelp -BuildConfig Debug "" exited with code 1.
```

**Comment by twitchax** [comment]
@azuresdkci test this please.

**Comment by cormacpayne** [comment]
@azuresdkci test this please

**Comment by twitchax** [comment]
@markcowl or @cormacpayne: what is causing this build to fail?

**Comment by twitchax** [comment]
@azuresdkci test this please.

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1530/

File: src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+---
+
+# Send-Feedback
+
+## SYNOPSIS
+Sends feedback to the Azure PowerShell team via a set of guided prompts.
+
+## SYNTAX
+
+### Default
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md]: Remove the Default Header

File: src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+```
+Send-Feedback
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md]: No common parameters in the syntax

File: src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+```
+
+## DESCRIPTION
+The Send-Feedback cmdlet sends feedback to the Azure PowerShell team.
+
+## EXAMPLES
+
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md]: I know this is simple, but can we have one example that shows screen output

File: src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
+---
```
Comment (markcowl) [inline src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md]: This did not make any change to: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Profile/Commands.Profile/help/AzureRM.Profile.md, if you geenrate help for the module using platyPS, this should be taken care of

Comment (twitchax) [inline src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md]: Done.

File: src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md (hunk 3 above)
Comment (twitchax) [inline src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md]: Done.

File: src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md (hunk 4 above)
Comment (twitchax) [inline src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md]: Done.

File: src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md (hunk 5 above)
Comment (twitchax) [inline src/ResourceManager/Profile/Commands.Profile/help/Send-Feedback.md]: Done.

**Comment by twitchax** [comment]
On demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1548/.

**Comment by twitchax** [comment]
@markcowl and @cormacpayne, fixed static analysis.

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/job/powershell-demand/1556/

**Pull request merged**